pyomo/contrib/doe/measurements.py

```diff
@@ -143,8 +147,11 @@ def _generate_variable_names_with_indices(
         added_names = []
         # iterate over index combinations ["CA", 1], ["CA", 2], ..., ["CC", 2], ["CC", 3]
         for index_instance in all_variable_indices:
-            var_name_index_string = var_name + "["
+            var_name_index_string = var_name
             for i, idx in enumerate(index_instance):
+                # if i is the first index, open the []
+                if i == 0:
+                    var_name_index_string += "["
```

Comment:
Can you re-use `name_repr` and `index_repr` from `pyomo.core.base.component_namer` here?

Reply:
Maybe. I added a comment to look into this as we refactor. I'm hoping the refactor makes this a moot point (as we move away from string manipulation).